### DISCUSSION TITLE
Version 2.0.0 Released

### DISCUSSION THREAD

**Discussion by temonk** [altasoft/DomainPrimitives#5] [Announcements]
Hello GitHub Community,

We're thrilled to announce the release of version 2.0.0 for AltaSoft.DomainPrimitives, and we've introduced some noteworthy changes that we want to share with you.

Firstly, we made a few breaking changes, such as removing the Altasoft.DomainPrimitives package, which has now been migrated to Altasoft.DomainPrimitives.XmlDataTypes. This restructuring was done to enhance the library's organization, focusing on more specific aspects like XmlDataTypes, which now covers a broader range of XML types compared to the previous Altasoft.DomainPrimitives package.

In addition to these changes, we've made significant improvements to our source generators. They are now faster, cache more data, and generate additional useful information. We've also incorporated support for XmlSerialization, addressing more developer needs. For projects using .NET 8, we've introduced the IUtf8SpanFormattable interface.

We've adjusted our recommendations for using string primitive types, now suggesting the use of class types. This decision was influenced by the fact that strings are treated as reference types, and employing structs for them introduced unnecessary complexity. Using classes simplifies the process, especially considering their default value is null.

Another exciting addition is the substantial enhancement to swagger type helpers. We've improved support for displaying default values of types in Swagger, and now, all types, including TimeOnly, Char, DateTimeOffset, and more, are fully supported. Additionally, we've enabled the ability to input an example as an XML comment, which will automatically be shown in Swagger. (Please see the example comment below)

![carbon (1)](https://github.com/altasoft/DomainPrimitives/assets/47372530/17c1ff90-7be9-4a35-b736-b7eca76f9e41)

We hope these updates enhance your experience with AltaSoft.DomainPrimitives, and we welcome any feedback or questions you may have. Happy coding!

**Comment by GregoryNikolaishvili** [comment]
Great news !!!

**Comment by MmixaA** [comment]
Great Work <3
